### PR TITLE
[17.0] [MIG] rma_reason_code

### PR DESCRIPTION
**Pull request by MeritxellAForgeFlow**
Migration to 17.0
Depends on #529

@ForgeFlow

### CODE REVIEW

File: rma_reason_code/models/reason_code.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+class RMAReasonCode(models.Model):
+    _name = "rma.reason.code"
+    _description = "RMA Reason Code"
+
+    def _get_default_color(self):
+        return randint(1, 11)
+
+    name = fields.Char("Code", required=True)
+    description = fields.Text()
+    type = fields.Selection(
```
Comment (MiquelRForgeFlow) [inline rma_reason_code/models/reason_code.py]: why not this? 
```suggestion
    something_type = fields.Selection(
```
it's bad trend to use the type field, as already exists [type](https://docs.python.org/3/library/functions.html#type) python function

File: rma_reason_code/reports/rma_reason_code_report.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+class RmaReasonCodeReport(models.Model):
+    _name = "rma.reason.code.report"
+    _auto = False
+    _description = "Rma Reason Code Report"
+
+    rma_order_line_id = fields.Many2one(comodel_name="rma.order.line")
+    reason_code_id = fields.Many2one(comodel_name="rma.reason.code")
+    date_rma = fields.Datetime(string="Order Date")
+    type = fields.Selection([("customer", "Customer"), ("supplier", "Supplier")])
```
Comment (MiquelRForgeFlow) [inline rma_reason_code/reports/rma_reason_code_report.py]: same here

**Comment by HviorForgeFlow** [comment]
Can remove https://github.com/ForgeFlow/stock-rma/pull/568/commits/735847b8d1b95fe4f5d304dd706cc32097acbb42

Seems code from another module

File: rma_reason_code/reports/rma_reason_code_report_views.xml [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+              </p><p>
+                Assign a Reason Code to a RTV
+              </p>
+          </field>
+      </record>
+
+      <menuitem
+        action="action_rma_reason_code_report"
+        id="menu_rma_reason_code_report"
+        parent="rma.menu_rma_rma_report"
```
Comment (MiquelRForgeFlow) [inline rma_reason_code/reports/rma_reason_code_report_views.xml:111]: it's used here @HviorForgeFlow

File: rma_reason_code/models/reason_code.py (hunk 1 above)
Comment (MeritxellAForgeFlow) [reply rma_reason_code/models/reason_code.py]: done!

**Review by MiquelRForgeFlow — APPROVED** [review]

**Review by HviorForgeFlow — APPROVED** [review]

File: rma_reason_code/models/reason_code.py [hunk 4]
```diff
@@ -13,8 +13,8 @@ def _get_default_color(self):
         return randint(1, 11)
 
     name = fields.Char("Code", required=True)
-    description = fields.Text("Description")
-    type = fields.Selection(
+    description = fields.Text()
+    item_type = fields.Selection(
```
Comment (LoisRForgeFlow) [inline rma_reason_code/models/reason_code.py]: now that we do rename the field, I think it is more accurate `rma_type`

Comment (MeritxellAForgeFlow) [reply rma_reason_code/models/reason_code.py]: done!

File: rma_reason_code/models/reason_code.py [hunk 5]
```diff
@@ -13,8 +13,8 @@ def _get_default_color(self):
         return randint(1, 11)
 
     name = fields.Char("Code", required=True)
-    description = fields.Text("Description")
-    type = fields.Selection(
+    description = fields.Text()
+    rma_type = fields.Selection(
```
Comment (LoisRForgeFlow) [inline rma_reason_code/models/reason_code.py:17]: add explicitly the string="RMA Type" in order to use uppercase in RMA.

image

File: rma_reason_code/views/rma_order_line_views.xml [hunk 6]
```diff
@@ -48,7 +48,7 @@
                     widget="many2many_tags"
                     options="{'color_field': 'color', 'no_create': True}"
                 />
-                <field name="allowed_reason_code_ids" invisible="1" />
+                <field name="allowed_reason_code_ids" column_invisible="1" />
```
Comment (LoisRForgeFlow) [inline rma_reason_code/views/rma_order_line_views.xml]: This is a form view, column_invisible only applies to tree/list views

image

**Review by LoisRForgeFlow — CHANGES_REQUESTED** [review]
Some comments below

File: rma_reason_code/models/reason_code.py (hunk 5 above)
Comment (MeritxellAForgeFlow) [reply rma_reason_code/models/reason_code.py:17]: done!

File: rma_reason_code/views/rma_order_line_views.xml (hunk 6 above)
Comment (MeritxellAForgeFlow) [reply rma_reason_code/views/rma_order_line_views.xml]: done x2!

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM now thanks!

**Pull request merged**